skyfield/positionlib.py

```diff
@@ -329,3 +344,27 @@ def ITRF_to_GCRS(jd, rITRF):  # todo: velocity
     spin = rot_z(jd.gast * TAU / 24.0)
     position = einsum('ij...,j...->i...', spin, array(rITRF))
     return einsum('ij...,j...->i...', jd.MT, position)
+
+
+def GCRS_to_Topos(xyz, jd):
+    dublin_julian_date = jd.tt - 2415020
+    x, y, z = xyz
+
+    # the following block is ported from GetSubSatPoint() in
+    # earthsat.c of libastro 3.7.6
+    sidereal_solar = 1.0027379093
+    sid_day = floor(dublin_julian_date)
+    t = (sid_day - 0.5) / 36525
+    sid_reference = (6.6460656 + (2400.051262 * t) +
+                     (0.00002581 * (t**2))) / 24
+    sid_reference -= floor(sid_reference)
```

Comment:
I am not sure — I may not have ever read this PyEphem code before — but this might be trying to compute, using an old and inexact formula, the sidereal time for the Julian Date in question? If so, you might want to take a look at the attribute `jd.gast` on the Julian Date object, which uses a modern IAU formula for the same thing. Try adding some `print` statements here that compare the numbers this old PyEphem formula is generating with the number that `.gast` produces and let me know if you see any resemblance. Thanks!